grafana/grafana
# Prometheus: Throw error on direct access

This is a copy of
- https://github.com/grafana/grafana/pull/49014
- https://github.com/grafana/grafana/pull/49246

Changes:
- Prometheus will throw an error on HTTP request when access == 'browser'.
- Config page will show error message instead of warning when access == 'browser' is selected.
- Config page will not show access select if access == 'proxy' as that is only supported mode going forward.

![Screenshot from 2022-06-03 13-45-22](https://user-images.githubusercontent.com/1014802/171857234-9177d18d-aa8f-40b9-8069-eabe8ccf94c6.png)

To test:
- Create a data source with access == 'browser' before checking out this PR.
- checkout PR
- Try to query data source -> should result in error
- Change the access to proxy/server
- Querying should work
- Going back to config page the select should not be visible anymore.

**Which issue(s) this PR fixes:**

Fixes #43859

# Release notice breaking change

In Prometheus, browser access mode was deprecated in Grafana 7.4.0 and removed in 9.2.0. If you used this mode, please switch to server access mode on the datasource configuration page.
